codegen/src/generator/ec2.rs

```diff
@@ -408,7 +410,7 @@ fn generate_struct_field_serializers(shape: &Shape) -> String {
                 ",
                 field_name = generate_field_name(member_name),
                 member_shape_name = member.shape,
-                tag_name = member_name,
+                tag_name = tag_name,
```

Comment:
Previously, this was always set to `member_name`, now it will be set to `member.location_name` first, unless it's `None`.

Is this logic change intentional?